packages/ckeditor5-coremedia-blocklist/src/blocklistui.ts

```diff
@@ -0,0 +1,443 @@
+
+const BLOCKLIST_KEYSTROKE = "Ctrl+Shift+B";
+
+/**
+ * The Blocklist UI plugin. It introduces the `'blocklist'` button and support for the <kbd>Ctrl+Shift+B</kbd> keystroke.
+ * It also creates the views, that are displayed inside the contextual balloon when triggered by the button, click or
+ * keystroke.
+ *
+ * It uses the contextual balloon plugin}.
+ */
+export default class Blocklistui extends Plugin {
+  static readonly pluginName: string = "BlocklistUI";
+  static readonly requires = [ContextualBalloon, BlocklistEditing];
+
+  blocklistActionsView: BlocklistActionsView;
+
+  blocklistCommand: BlocklistCommand | undefined;
+
+  #balloon: ContextualBalloon | undefined = undefined;
+
+  constructor(editor: Editor) {
+    super(editor);
+    this.blocklistActionsView = this.#createBlocklistActionsView();
+  }
+
+  async init(): Promise<void> {
+    const editor = this.editor;
+    this.blocklistCommand = (await ifCommand(editor, BLOCKLIST_COMMAND_NAME)) as BlocklistCommand;
+
+    this.#balloon = editor.plugins.get(ContextualBalloon);
+
+    // listen to click and key events to open the blocklist balloon
+    this.#initBalloonListeners();
+
+    // adds a blocklist button to the componentFactory
+    this.#createBlocklistToolbarButton(this.blocklistCommand);
+
+    // listen to click and key events while the balloon is open (to navigate or close the balloon)
+    this.#initBalloonViewListeners();
+
+    this.blocklistActionsView.bind("blockedWords").to(this.blocklistCommand, "value");
+
+    // listen to changes in blocklistCommand and refresh the list in the blocklist view accordingly
+    this.blocklistCommand.on("change:value", this.blocklistActionsView.refreshList.bind(this.blocklistActionsView));
+  }
+
+  /**
+   * Creates the blocklist view, that will be displayed inside a contextual balloon.
+   * Also registers a listener to close the view on ESC when the view is focused.
+   *
+   * @private
+   */
+  #createBlocklistActionsView(): BlocklistActionsView {
+    const editor = this.editor;
+
+    const blocklistActionsView = new BlocklistActionsView(editor);
+
+    this.listenTo(blocklistActionsView.blocklistInputView, "submit", () => {
+      const blockWordInput = this.blocklistActionsView.blocklistInputView.getInputElement();
+
+      if (!this.blocklistCommand) {
+        return;
+      }
+
+      // Return, if the word is already in the blocklist
+      const blacklistCommandValue = this.blocklistCommand.value;
+      if (blacklistCommandValue.includes(blockWordInput.value)) {
+        return;
+      }
+
+      // Add markers and sync with service
+      const editingPlugin = editor.plugins.get(BlocklistEditing);
+      editingPlugin.addBlocklistWord(blockWordInput.value);
+
+      const newValue = [...blacklistCommandValue, blockWordInput.value];
+      this.blocklistCommand.set("value", newValue);
+
+      // Clear the input
+      this.blocklistActionsView.blocklistInputView.setInputText("");
```

Comment:
As discussed in the pair review:
Accesses the class field `this.blocklistActionsView` but the listener is defined in the function which create the view and afterwards assigns to the field which made me stumble while reading the code.

Maybe it would be better to use the function variable blocklistActionsView.